web/source/build.sh

```diff
@@ -0,0 +1,233 @@
+#! /bin/bash
+# 
+# Compile keymanweb and copy compiled javascript and resources to output/embedded folder
+#
+
+display_usage ( ) {
+    echo "build.sh [-ui | -test | -embed | -web | -debug_embedded]"
+    echo
+    echo "  -ui               to compile desktop user interface modules to output folder"
+    echo "  -test             to compile for testing without copying resources or" 
+    echo "                    updating the saved version number."
+    echo "  -embed            to compile only the KMEA/KMEI embedded engine."
+    echo "  -web              to compile only the KeymanWeb engine."
+    echo "  -debug_embedded   to compile a readable version of the embedded KMEA/KMEI code"
+    exit 1
+}
+
+# Fails the build if a specified file does not exist.
+assert ( ) {
+    if ! [ -f $1 ]; then
+        echo "Build failed."
+        exit 1
+    fi
+}
+
+# Definition of global compile constants
+WEB_OUTPUT="../output"
+EMBED_OUTPUT="../embedded"
+SOURCE="."
+
+readonly WEB_OUTPUT
+readonly EMBED_OUTPUT
+readonly SOURCE
```

Comment:
Every example shows them being two separate lines, and apparently that pattern can be prone to errors at times.